src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppAppRunner/Configurations/Configuration.cs

```diff
@@ -0,0 +1,120 @@
+        /// </summary>
+        public int? HealthCheckInterval { get; set; }
+
+        /// <summary>
+        /// The URL that health check requests are sent to.
+        /// </summary>
+        public string? HealthCheckPath { get; set; }
+
+        /// <summary>
+        /// The IP protocol that App Runner uses to perform health checks for your service.
+        /// </summary>
+        public string HealthCheckProtocol { get; set; } = "TCP";
+
+        /// <summary>
+        /// The time, in seconds, to wait for a health check response before deciding it failed.
+        /// </summary>
+        public int? HealthCheckTimeout { get; set; }
+
+        /// <summary>
+        /// The number of consecutive checks that must fail before App Runner decides that the service is unhealthy.
+        /// </summary>
+        public int? HealthCheckUnhealthyThreshold { get; set; }
+
+        /// <summary>
+        /// The number of CPU units reserved for each instance of your App Runner service.
+        /// </summary>
+        public string Cpu { get; set; }
+
+        /// <summary>
+        /// The amount of memory, in MB or GB, reserved for each instance of your App Runner service.
+        /// </summary>
+        public string Memory { get; set; }
+
+        /// A parameterless constructor is needed for <see cref="Microsoft.Extensions.Configuration.ConfigurationBuilder"/>
+        /// or the classes will fail to initialize.
+        /// The warnings are disabled since a parameterless constructor will allow non-nullable properties to be initialized with null values.
+#nullable disable warnings
+        public Configuration()
+        {
+
+        }
+#nullable restore warnings
+
+        public Configuration(
+            IAMRoleConfiguration applicationIAMRole,
+            IAMRoleConfiguration serviceAccessIAMRole,
+            string serviceName,
+            int? port,
+            string? startCommand,
+            string? encryptionKmsKey,
+            double? healthCheckHealthyThreshold,
+            int? healthCheckInterval,
+            string? healthCheckPath,
+            string healthCheckProtocol,
+            int? healthCheckTimeout,
+            int? healthCheckUnhealthyThreshold,
+            string cpu,
+            string memory)
+        {
+            ApplicationIAMRole = applicationIAMRole;
+            ServiceAccessIAMRole = serviceAccessIAMRole;
+            ServiceName = serviceName;
+            Port = port ?? 80;
+            StartCommand = startCommand;
+            EncryptionKmsKey = encryptionKmsKey;
+            HealthCheckHealthyThreshold = healthCheckHealthyThreshold;
+            HealthCheckInterval = healthCheckInterval;
+            HealthCheckPath = healthCheckPath;
+            HealthCheckProtocol = healthCheckProtocol;
+            HealthCheckTimeout = healthCheckTimeout;
+            HealthCheckUnhealthyThreshold = healthCheckUnhealthyThreshold;
+            Cpu = cpu;
+            Memory = memory;
+        }
```

Comment:
nit: also for consistency, we are not including the nullable properties as part of the constructor.

Reply:
Removed except I left port because the constructor ensures the port is at least set to 80